pallets/parachain-staking/src/lib.rs

```diff
@@ -513,6 +513,11 @@ pub mod pallet {
 		ValueQuery,
 	>;
 
+	#[pallet::storage]
+	#[pallet::getter(fn leaving_delegators)]
+	/// List of delegators scheduled for leave
+	pub(crate) type LeavingDelegators<T: Config> = StorageValue<_, Vec<T::AccountId>, ValueQuery>;
```

Comment:
This a single key storage item, so it means each interaction is reading/writing it. In the worst case (very unlikely) it is a lot of data to move for each extrinsic call.

Would it be better to consider the leave as simply request all delegations as revoked, and the execute leave to simply execute all those revoke requests ?

Reply:
> Would it be better to consider the leave as simply request all delegations as revoked, and the execute leave to simply execute all those revoke requests ?
From the perspective of functionality - it might create side-effects since we'd be mixing it with the `Revoke` flow that means anyone can cancel a single revoke and thus invalidate the `Leave`. 

#1515 mitigates that with a new `Leave` variant that is uniquely manipulated with leave request. It's not the most straightforward solution but is certainly the most performant one.

Reply:
It's very interesting, if I understand correctly, we will read less storage for the delegators who leave, because we don't need to access their balance to pay them rewards.
So we can afford to add a storage item to store the leaves :)